src/nvim/path.c

```diff
@@ -2186,9 +2186,16 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf,
 
 /// Check if the given file is absolute.
 ///
-/// This just checks if the file name starts with '/' or '~'.
 /// @return `TRUE` if "fname" is absolute.
 int path_is_absolute_path(const char_u *fname)
 {
+#ifdef WIN32
+  // A name like "d:/foo" and "//server/share" is absolute
+  return (isalpha(fname[0]) && fname[1] == ':'
```

Comment:
AFAIR `isalpha` is locale-dependent and may return true unexpectedly if locale happens to be non-UTF-8 (and, theoretically, it may return unexpected false if locale is not `C`, but I doubt you will find any locale which rejects latin letters). I would suggest using not C library, but our `ascii_isalpha` function from `src/nvim/lib/ascii.h`.